src/engine.ml

```diff
@@ -357,7 +357,15 @@ let maybe_kex_client rng config tls =
     let pre_master, random1, random2 = (rng 48, rng 32, rng 32) in
     Config.client_generate_connect_options config >>= fun options ->
     let user_pass = Config.find Auth_user_pass config in
-    let td = { Packet.pre_master; random1; random2; options; user_pass }
+    let peer_info =
+      let ciphers =
+        String.concat ":"
+          (List.map Config.cipher_to_string (Config.get Data_ciphers config))
+      in
+      Some [ "IV_PLAT=mirage"; "IV_CIPHERS=" ^ ciphers; "IV_NCP=2" ]
```

Comment:
should there be more to it? should it be confgurable whether we do NCP or not (i.e. --disable-ncp)? or is that overthinking?

Reply:
Maybe we can send as well `IV_PROTO=0`

Reply:
Or maybe it could be interesting with `IV_PROTO=4` signalling "[t]he client expects a push-reply and  the  server may  send this reply without waiting for a push-request first."

Reply:
hmm hmm, but we'll send a push_request!? but eventually this is not needed and would save a roundtrip?

Reply:
Several options are set if we want to save a roundtrip according to what `openvpn` does:
https://github.com/OpenVPN/openvpn/blob/7c637b35038e4cc7d2f16e07f413cfa2b0f1f97a/src/openvpn/ssl.c#L2054-L2070

Reply:
So, it's more `IV_PROTO=276` (`REQUEST_PUSH` | `AUTH_PENDING_KW` | `AUTH_FAIL_TEMP`) plus `IV_MTU` than `IV_PROTO=4`

Reply:
thanks for the link, I think for now, the `REQUEST_PUSH` is the only one we can achieve -- since we do not handle temporary authentication failures, neither auth_pending control messages.

the `IV_MTU` we should sent as well (which is a separate key-value from `IV_PROTO`).

Reply:
I opened #126 to not forget about this

Reply:
For compatibility with openvpn 2.4 `IV_NCP=2` signals the client supports AES-{128,256}-GCM and is deprecated in favor of `IV_CIPHERS` according to https://github.com/OpenVPN/openvpn-rfc/blob/bbde60d7f28a5436e21b3f475788f7922e0d9339/openvpn-wire-protocol.xml#L1398-L1399

We can deal with that in another PR as this is already a lot of changes.